### PR TITLE
feat: schedule form closing date

### PR DESCRIPTION
**Pull request by thiessenp-cds**
# Summary | Résumé

Allows a user to schedule a future date to close a form.

**The schedule a date dialog is now behind a feature flag. Remember to enable "scheduleClosingDate" for testing locally or on staging.**

## Test

Open a form and go to the Settings > Manage tab. 
Click the "Schedule form closing" link. A dialog should appear. Try setting a date. The Manage screen should update with the schedule date. Also when viewing the form, there should be a banner at the top of the form with a close date.

Then try to break it. In the dialog, try setting a date in the past, adding invalid inputs.. Try toggling the the form to open/close etc. just see if you can break it :)

### CODE REVIEW

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/close/SetClosingDate.tsx [hunk 1]
```diff
@@ -44,16 +47,44 @@ export const SetClosingDate = ({
...
+      setClosingDate(closingDate);
+
+      toast.success(t("closingDate.savedSuccessMessage"));
+    },
+    [formId, setClosingDate, t, closedMessage]
+  );
+
   const saveFormStatus = useCallback(async () => {
-    let closeDate = "open";
+    let closeDate = "open"; // this wil reset the closing date to null;
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/close/SetClosingDate.tsx:84]: We can pass just null here and update the server action but we need to check of the closing date is alresdy set in the futrure before we null out the date.

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/close/SetClosingDate.tsx [hunk 2]
```diff
@@ -46,7 +46,17 @@ export const SetClosingDate = ({
     return true;
   }, [closedMessage]);
 
-  const [status, setStatus] = useState(closingDate ? "closed" : "open");
+  // TODO: not working, needs more work
+  const isPastDate = useCallback(
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/close/SetClosingDate.tsx:50]: Can use the dateHasPast util func here
```
const isPastClosingDate = dateHasPast(Date.parse(closingDate));
```

File: i18n/translations/en/form-builder.json [hunk 3]
```diff
@@ -1158,5 +1158,24 @@
         "description": "Add a relevant action that is specific for the people filling out the form. For example, “Remove address”"
       }
     }
+  },
+  "scheduleClosingPage": {
+    "linkText": "Schedule closing page",
```
Comment (anikbrazeau) [inline i18n/translations/en/form-builder.json:1163]:
```suggestion
    "linkText": "Schedule form closing",
```

File: i18n/translations/en/form-builder.json [hunk 4]
```diff
@@ -1158,5 +1158,24 @@
...
+      "text1": "Schedule a closing date",
+      "text2": "Set when you’d like to close the form.",
+      "datePicker": {
+        "month": "Month",
+        "day": "Day",
+        "year": "Year"
+      },
+      "timePicker": {
+        "text1": "Set a time",
+        "text2": "Provide time in 24 hour clock. The timezone is in Eastern time"
```
Comment (anikbrazeau) [inline i18n/translations/en/form-builder.json:1175]:
```suggestion
        "text2": "Provide the time using the 24-hour clock with timezone in Eastern Time"
```

File: i18n/translations/fr/form-builder.json [hunk 5]
```diff
@@ -1158,5 +1158,24 @@
...
+        "day": "Day [FR]",
+        "year": "Year [FR]"
+      },
+      "timePicker": {
+        "text1": "Set a time [FR]",
+        "text2": "Provide time in 24 hour clock. The timezone is in Eastern time [FR]"
+      },
+      "save": "Save [FR]",
+      "cancel": "Cancel [FR]"
+    }
```
Comment (anikbrazeau) [inline i18n/translations/fr/form-builder.json:1179]:
```suggestion
  },
  "scheduleClosingPage": {
    "linkText": "Programmer la fermeture du formulaire",
    "dialog": {
      "title": "Programmez le message de fermeture du formulaire",
      "text1": "Fixez la date de fermeture",
      "text2": "Décidez la date à laquelle vous souhaitez fermer le formulaire",
      "datePicker": {
        "month": "Mois",
        "day": "Jour",
        "year": "Année"
      },
      "timePicker": {
        "text1": "Fixez l'heure de fermeture",
        "text2": "Indiquez l'heure en utilisant l'horloge de 24 heures avec le fuseau horaire de l'heure de l'Est"
      },
      "save": "Enregistrer",
      "cancel": "Annuler"
    }
```

File: i18n/translations/fr/form-builder.json [hunk 6]
```diff
@@ -1158,5 +1158,24 @@
...
+  "scheduleClosingPage": {
+    "linkText": "Programmer la fermeture du formulaire",
+    "dialog": {
+      "title": "Programmez le message de fermeture du formulaire",
+      "text1": "Fixez la date de fermeture",
+      "text2": "Décidez la date à laquelle vous souhaitez fermer le formulaire",
+      "datePicker": {
+        "month": "Mois",
+        "day": "Jour",
+        "year": "Année"
```
Comment (anikbrazeau) [inline i18n/translations/fr/form-builder.json:1171]: In French, this should be in a different order:

[Day] [Month] [Year]

Comment (timarney) [reply i18n/translations/fr/form-builder.json:1194]: Pete updated the UI.

**Review by timarney — APPROVED** [review]

**Pull request merged**